pkg/factsengine/entities/fact_value.go

```diff
@@ -44,20 +57,34 @@ func NewFactValue(factInterface interface{}) (FactValue, error) {
 	case map[string]interface{}:
 		newMap := make(map[string]FactValue)
 		for key, mapValue := range value {
-			newValue, err := NewFactValue(mapValue)
+			newValue, err := NewFactValueWithConf(mapValue, conf)
 			if err != nil {
 				return nil, err
 			}
+			if conf.SnakeCaseKeys {
+				key = strcase.ToSnake(key)
+			}
 			newMap[key] = newValue
 		}
 		return &FactValueMap{Value: newMap}, nil
-	case bool, int, int32, int64, uint, uint32, uint64, float32, float64, string:
+	case bool, int, int32, int64, uint, uint32, uint64, float32, float64:
 		return ParseStringToFactValue(fmt.Sprint(value)), nil
+	case string:
+		if conf.StringConversion {
+			return ParseStringToFactValue(value), nil
+		}
+		return &FactValueString{Value: value}, nil
 	default:
 		return nil, fmt.Errorf("invalid type: %T for value: %v", value, factInterface)
 	}
 }
 
+// NewFactValue constructs a FactValue from a nested interface.
+func NewFactValue(factInterface interface{}) (FactValue, error) {
```

Comment:
Could we have this constructor named `NewFactValueWithDefaultConf`? In this way we can explicitely say that we are using this constructor as a proxy with the new factValue with the default configuration.

We don't have real optional values in golang and having this naming conventions helps us to recognize this proxies.

Reply:
Initially I didn't want to do that because:
- This is "somehow" a public API, as it it is in the `package` folder, and it could be maybe used by some external user (on a plugin?). Anyway, i really doubt this
- It would force me to change the current usages of the code.

None of them a big deal. I will change.
Just writing down the rationale about this implementation

Reply:
I have finally used the functional options pattern, which allows us to have the same signature.
It affect minimally to the usage, as we only have one default option by now (the StringConversion)